### PR TITLE
Aiming with arrows ( + basis for alpha value)

### PR DESCRIPTION
**Pull request by nurgan**
Aiming with arrows works now. 

Created the basis for using alpha values with our materials, if we want to at some point. To actually use it, would need objects with alpha being drawn last, and to be sorted depended on the players z.

### CODE REVIEW

File: src/CookieActionComponent.cpp [hunk 1]
```diff
@@ -170,37 +170,62 @@ void CookieActionComponent::checkAndPerformAction(double deltaTime, double total
...
     }
+
+}
+
+void CookieActionComponent::setAimRotation(std::shared_ptr <GameObject> obj, GameManager& gameManager) {
+    std::shared_ptr<GameObject> player = gameManager.getPlayer();
+    float playerRot = player->getYAxisRotation() - player->getOrientAngle() + yRot;
+
+    obj->setYAxisRotation(playerRot);
+    obj->addRotation(xRot, glm::vec3(0.0, 0.0, -1.0));
```
Comment (nurgan) [inline src/CookieActionComponent.cpp:230]: tbh, I have no idea why I have to rotate around the negative z axis, but it works.

File: resources/phong_frag.glsl [hunk 2]
```diff
@@ -205,5 +206,5 @@ void main() {
 
 
 	// Calculate the total color
-	color = shadowFactor * vec4(directionalLightColor, 1.0);
+	color = shadowFactor * vec4(directionalLightColor, MatAlpha);
```
Comment (rgarmsen2295) [inline resources/phong_frag.glsl:209]: Ooo very cool

File: src/CookieActionComponent.cpp (hunk 1 above)
Comment (rgarmsen2295) [reply src/CookieActionComponent.cpp:230]: kk lol sounds good

**Review by rgarmsen2295 — CHANGES_REQUESTED** [review]
All looks good but please add the MatAlpha to the two other core shaders (Toon and CookTorr) so that it still looks good there as well.  Otherwise, code looks great!

**Review by joncatanio — COMMENTED** [review]
This looks awesome, once you fix the changes that Reed requested go ahead and merge. What do you guys think on changing the yellow color though? Maybe start with white?

**Comment by nurgan** [comment]
I'll merge it for now, and play with colors later, I wanna get all the PRs merged

**Pull request merged**